tls/s2n_cipher_suites.c

```diff
@@ -1168,11 +1171,158 @@ static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, u
     return 0;
 }
 
+S2N_RESULT s2n_cipher_preferences_contain(const struct s2n_cipher_preferences *cipher_preferences, uint8_t iana[S2N_TLS_CIPHER_SUITE_LEN], bool *cipher_preferences_contain) {
+    RESULT_ENSURE_REF(cipher_preferences);
+    RESULT_ENSURE_REF(cipher_preferences_contain);
+
+    *cipher_preferences_contain = false;
+
+    for (int i = 0; i < cipher_preferences->count; i++) {
+        const uint8_t *cipher_pref_iana = cipher_preferences->suites[i]->iana_value;
+        if (s2n_constant_time_equals(cipher_pref_iana, iana, S2N_TLS_CIPHER_SUITE_LEN)) {
+            *cipher_preferences_contain = true;
+            return S2N_RESULT_OK;
+        }
+    }
+
+    return S2N_RESULT_OK;
+}
+
+S2N_RESULT s2n_iana_is_a_chacha20_cipher_suite(const uint8_t iana[S2N_TLS_CIPHER_SUITE_LEN], bool* is_chacha20_cipher_suite) {
+    RESULT_ENSURE_REF(is_chacha20_cipher_suite);
+
+    *is_chacha20_cipher_suite = false;
+    if (s2n_constant_time_equals(s2n_tls13_chacha20_poly1305_sha256.iana_value, iana, S2N_TLS_CIPHER_SUITE_LEN) ||
+        s2n_constant_time_equals(s2n_ecdhe_rsa_with_chacha20_poly1305_sha256.iana_value, iana, S2N_TLS_CIPHER_SUITE_LEN) ||
+        s2n_constant_time_equals(s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256.iana_value, iana, S2N_TLS_CIPHER_SUITE_LEN) ||
+        s2n_constant_time_equals(s2n_dhe_rsa_with_chacha20_poly1305_sha256.iana_value, iana, S2N_TLS_CIPHER_SUITE_LEN)) {
+        
```

Comment:
This list is going to be very hard to keep up to date. You should really be checking the record alg of the cipher suite structure, not hard-coding a list. But that's going to be difficult if you don't have a cipher suite structure to compare to, so might require some thinking about how you're handling this logic.